packages/flutter_tools/templates/create/android.tmpl/buildSrc/src/main/groovy/FlutterPlugin.groovy

```diff
@@ -0,0 +1,229 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
```

Comment:
The FlutterPlugin is the code that hooks us into the build, by creating a FlutterTask (that builds the Dart code and zip up the assets), and wire up the dependencies in the build DAG. There are ways in Gradle to depend on external plugins by pointing at an artifact in a Maven repository, but we're special, so we need to tweak it a little.

I'm trying to avoid having flutter tools do too much other than invoke gradle at build time. Ideally, this should also work as a standalone Gradle project, so we can just open it in Android Studio and hit play, but it's probably fine to require an initial build using flutter tools that can set the path to the plugin correctly.